src/program.ts

```diff
@@ -671,6 +671,48 @@ export class Program extends DiagnosticEmitter {
     );
   }
 
+  // TODO: make these more generic, purpose built for closures atm
+  makeNativeMember(
```

Comment:
Do these helpers exist to construct actual source-level classes backing closure contexts here? If so, that seems like unnecessary overhead since an actual class doesn't give us anything, like a user interacting with it or proper diagnostics. Suggesting to use a compiler-level simple class instead holding the necessary state, completely independent of sources.

Reply:
By compiler-level simple class do you mean just constructing a class manually? We certainly could, just means we don't get to reuse all the struct creation logic which determines offsets and such that exists within compileClass. But if we're worried about the overhead I will look into it